pkg/parse/run.go

```diff
@@ -234,6 +234,12 @@ func run(ctx context.Context, p Parser, trigger string, state *reconcilerState)
 // It also updates the .status.rendering and .status.source fields.
 func read(ctx context.Context, p Parser, trigger string, state *reconcilerState, sourceState sourceState) status.MultiError {
 	hydrationStatus, sourceStatus := readFromSource(ctx, p, trigger, state, sourceState)
+	if status.HasTransientErrors(hydrationStatus.errs) {
+		return hydrationStatus.errs
```

Comment:
Will the other non-transient errors still get updated into the source status?

Reply:
If `hydrationStatus.errs` has other non-transient errors, it won't get updated into the source status in the current loop, but will get retried.

But on the other hand, for now, if it has transient errors, it is unlikely to have other errors because the current known transient errors are:
- unable to read the error file error because *-sync container is not ready yet.
- the race condition you're fixing.
They'll terminate the current parse-apply-watch loop, and retry.

Reply:
Makes sense!